### ISSUE TITLE
num-layers issue

### ISSUE THREAD

**Issue by Bugercreater** [vllm-project/speculators#238]
I modified num-layers to 2 while training a spec model, loss turns to nan from the first step. But when num-layers was 1 it works well. 

log:
[14:42:04] INFO     train/loss_0=nan, train/full_acc_0=0.002, train/cond_acc_0=0.002, train/loss_1=nan, train/full_acc_1=0.00e+00, train/cond_acc_1=0.00e+00, train/loss_2=nan, train/full_acc_2=0.00e+00, train/cond_acc_2=0.00e+00, 

Do I need to change anything when I generate data or any other part of this training?

**Comment by dsikka** [comment]
@fynnsu

**Comment by olisicky** [comment]
Hi @fynnsu . Thank you looking into my problem. I am generally struggling to train draft model from more than 2 layers (loss is nan from first step). I can see some accuracy improvements going from 1 to 2 and would like to go further. I am training speculator for fine-tuned llama3.1-70B model. The hidden states and vocabulary mapping were generated with `data_generation_offline.py` and `build_vocab_mapping.py` respectively. Draft vocab size `128256` as target model. Training on 8xH200.

The training setup is here:

```
torchrun --nnodes=1 --nproc_per_node=${WORLD_SIZE}  scripts/train.py \
    --verifier-name-or-path ${BASE_MODEL} \
    --data-path "${BASE_DIR}/training_data" \
    --epochs 20 \
    --lr 5e-5 \
    --no-resume-from-checkpoint \
    --logger "tensorboard" \
    --total-seq-len 16384 \
    --data-format-version 1 \
    --log-dir "${BASE_DIR}/logs/${MODEL_NAME}" \
    --run-name "${MODEL_NAME}" \
    --num-layers 3 \
    --d2t-path "${BASE_DIR}/vocab_mapping/d2t.npy" \
    --t2d-path "${BASE_DIR}/vocab_mapping/t2d.npy" \
    --ttt-steps 3 \
    --save-path "${BASE_DIR}/checkpoints" \
    --ttt-step-loss-decay 1.0 
```

Lib versions:
```
vllm==0.13.0
transformers==4.57.6
torch==2.9.0
tokenizers==0.22.2
speculators==0.4.0.dev36
cuda 12.9
```
